src/page-modules/assistant/server/journey-planner/index.ts

```diff
@@ -0,0 +1,265 @@
+import { GraphQlRequester } from '@atb/modules/api-server';
+import {
+  TripsDocument,
+  TripsQuery,
+  TripsQueryVariables,
+} from '@atb/page-modules/assistant/server/journey-planner/journey-gql/trip.generated';
+import {
+  Notice,
+  Situation,
+  TripData,
+  tripSchema,
+} from '@atb/page-modules/assistant/server/journey-planner/validators';
+import {
+  getTransportModesEnums,
+  isTransportModeType,
+  isTransportSubmodeType,
+} from '@atb/page-modules/departures/server/journey-planner';
+import { TripInput } from '@atb/page-modules/assistant';
+
+export type JourneyPlannerApi = {
+  trip(input: TripInput): Promise<TripData>;
+};
+
+export function createJourneyApi(
+  client: GraphQlRequester<'graphql-journeyPlanner3'>,
+): JourneyPlannerApi {
+  const api: JourneyPlannerApi = {
+    async trip(input) {
+      const modes = input.transportModes
+        ? {
+            transportModes: getTransportModesEnums(input.transportModes)?.map(
+              (mode) => ({
+                transportMode: mode,
+              }),
+            ),
+          }
+        : undefined;
+
+      const from = {
+        place: input.from.id,
+        coordinates: {
+          latitude: input.from.geometry.coordinates[1],
+          longitude: input.from.geometry.coordinates[0],
+        },
+        name: input.from.name,
+      };
+      const to = {
+        place: input.to.id,
+        coordinates: {
+          latitude: input.to.geometry.coordinates[1],
+          longitude: input.to.geometry.coordinates[0],
+        },
+        name: input.to.name,
+      };
+
+      const result = await client.query<TripsQuery, TripsQueryVariables>({
+        query: TripsDocument,
+        variables: {
+          from,
+          to,
+          arriveBy: input.arriveBy,
+          when: input.when,
+          numTripPatterns: 10,
+          modes,
+        },
+      });
+
+      if (result.error) {
+        throw result.error;
+      }
+
+      const data: RecursivePartial<TripData> = {
```

Comment:
I assume this kind of parsing can't be done with zod?

Reply:
I haven't found anything from their documentation that suggests so 🤔

Reply:
I figured. 👍